mars_troughs/accumulation_model.py

```diff
@@ -31,41 +28,203 @@ class InsolationAccumulationModel(AccumulationModel):
+        as in dx/dt=(R(l(t),t)+A(ins(t))cos(theta))/sin(theta). Where theta
+        is the angle of the trough.
+
+        Args:
+            time (np.ndarray): times at which we want the path.
+        Output:
+            horizontal distances (np.ndarray) of the same size as time input.
+        """
+        self.angle=angle           
+        yt = self.get_yt(time)
+        
+        return -self.cot_angle * yt + self.csc_angle * (
+         iretreat_model_t_spline(time) - iretreat_model_t_spline(0))
+    
+    @property
+    def angle(self) -> float:
+        """
+        Slope angle in degrees.
+        """
+        return self._angle * 180.0 / np.pi
+
+    @angle.setter
+    def angle(self, value: float) -> float:
+        """Setter for the angle"""
+        self._angle = value * np.pi / 180.0
+        self._csc = 1.0 / np.sin(self._angle)
+        self._cot = np.cos(self._angle) * self._csc
 
+    @property
+    def csc_angle(self) -> float:
+        """
+        Cosecant of the slope angle.
+        """
+        return self._csc
 
-class LinearInsolationAccumulation(InsolationAccumulationModel):
+    @property
+    def cot_angle(self) -> float:
+        """
+        Cotangent of the slope angle.
+        """
+        return self._cot
+
+class LinearAccuIns(AccumulationDepInsolation):
     """
     Accumulation is linear in solar insolation.
 
     Args:
         times (np.ndarray): times at which the solar insolation is known
         insolations (np.ndarray): value of the solar insolations
-        intercept (float, optional): default is 0 millimeters
-        slope (float, optional): default is 1e-6 mm per unit
-            of solar insolation square.
+        intercept (float, optional): default is 0 m
+        slope1 (float, optional): default is 1e-6 m per unit
+            of solar insolation.
     """
+    def __init__(self, times, insolation, intercept: float = 0.0, slope: float = 1e-6):
+        
+        super().__init__(times, insolation)
+        
+        self.intercept=intercept
+        self.slope=slope
+        
+    def get_accumulation_at_t(self, time: np.ndarray):
+        """
+        Accumulation rate A (in m/year) as a  linear function of insolation: 
+        A(ins(t)) = intercept + slope*ins(t).  Intercept in m/year and slope in 
+        m^3/(year*W).
 
-    def __init__(
-        self,
-        times: np.ndarray,
-        insolations: np.ndarray,
-        intercept: float = 1.0,
-        slope: float = 1e-6,
-    ):
-        super().__init__(times, insolations)
-        self.intercept = intercept
-        self.slope = slope
+        Args:
+            time (np.ndarray): times at which we want to calculate A, in years.
+        Output:
+            np.ndarray of the same size as time input containing values of 
+            accumulation rates A in m/year
+        
+        """
+        return self.intercept -1 * (self.slope * self.ins_data_spline(time))
```

Comment:
RE the sign issues discussed this morning - if you remove the `-1` here the accumulations will all look positive.

Reply:
done